data/nlmchem/transform.py

```diff
@@ -0,0 +1,103 @@
+import urllib.request
+import zipfile
+
+import pandas as pd
+import yaml
+
+
+def get_and_transform_data():
+    url = "https://ftp.ncbi.nlm.nih.gov/pub/lu/NLMChem/NLM-Chem-corpus.zip"
+    local_zip_path = "NLM-Chem-corpus.zip"
+
+    # Download the ZIP file
+    urllib.request.urlretrieve(url, local_zip_path)
+
+    # Open the ZIP file and extract the TSV file
+    with zipfile.ZipFile(local_zip_path, "r") as z:
+        with z.open("FINAL_v1/abbreviations.tsv") as f:
+            df = pd.read_csv(f, sep="\t", header=None)
+
+    # Set column names
+    df.columns = ["MeSH_Identifier", "Abbreviation", "Full_Form"]
+
+    # Remove duplicates
+    df.drop_duplicates(inplace=True)
+
+    # Check duplicates
+    assert not df.duplicated().sum(), "Found duplicate rows in the dataframe"
+
+    # Save to CSV
+    fn_data_csv = "nlmchem_data_clean.csv"
+    df.to_csv(fn_data_csv, index=False)
+
+    # Create meta.yaml
+    meta = {
+        "name": "NLM-Chem",
+        "description": (
+            "NLM-Chem is a new resource for chemical entity recognition in PubMed full text literature."
+        ),
+        "targets": [
+            {
+                "id": "MeSH_Identifier",
+                "description": "Unique codes for Medical Subject Headings",
+                "type": "categorical",
+                "names": ["MeSH_Identifier"],
+            },
+            {
+                "id": "Abbreviation",
+                "description": "Abbreviation of a term",
+                "type": "categorical",
+                "names": ["Abbreviation"],
+            },
+            {
+                "id": "Full_Form",
+                "description": "Full form or meaning of the abbreviation",
+                "type": "categorical",
+                "names": ["Full_Form"],
+            },
+        ],
+        "license": "CC BY 4.0",
+        "links": [
+            {
+                "url": "https://ftp.ncbi.nlm.nih.gov/pub/lu/NLMChem/",
+                "description": "data source",
+            },
+            {
+                "url": "https://www.nature.com/articles/s41597-021-00875-1",
+                "description": "publication",
+            },
+        ],
+        "num_points": len(df),
+        "bibtex": """
```

Comment:
Format the bibtex entry like shown here https://github.com/OpenBioML/chemnlp/blob/main/data/caco2_wang/transform.py#L110 to avoid the newline breaks.